### PR TITLE
Add VADR to flu and RSV TheiaCoV

### LINKED ISSUE #295: Adding flu and RSV to TheiaCoV fastq workflows (VADR)
:cool:

### :pushpin: Explain the Request

Update the TheiaCoV fastq workflows (e.g. Illumina PE/SE, ONT, etc) to add VADR to influenza and RSV organism tracks, similar to [TheiaCoV_FASTA](https://github.com/theiagen/public_health_bioinformatics/blob/main/workflows/theiacov/wf_theiacov_fasta.wdl#L96)

### :books: Context

Will help laboratories perform QC of additional viral assemblies generated

### :chart_with_upwards_trend: Desired Behavior

VADR outputs produced for more viral pathogens

### PR DESCRIPTION
**Pull request by cimendes**
This PR closes #295 

🗑️ This dev branch should <NOT> be deleted after merging to main.

## :brain: Aim, Context and Functionality

This PR adds VADR to two new organisms:
- Flu
- RSV

Due to the limitations of the RSV model the VADR memory parameter has been exposed in the organisms parameter sub-workflow. 

Additionally, the VADR container has been upgraded to the latest version: `us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.3`

## :hammer_and_wrench:  Impacted Workflows/Tasks & Changes Being Made
This will affect the behavior of the workflow(s) even if users don’t change any workflow inputs relative to the last version : Yes (Docker container has been updated)

Running this workflow on different occasions could result in different results, e.g. due to use of a live database, "latest" docker image, or stochastic data processing : No 

#### 🔄 Data Processing 

Docker/software or software versions changed: Yes, changed to `us-docker.pkg.dev/general-theiagen/staphb/vadr:1.6.3`

Databases or database versions changed: N/A

Data processing/commands changed: N/A

File processing changed: N/A

Compute resources changed: Memory for VADR has been adjusted depending on the organism (8 for all organisms except for RSV-A and RSV-B which is set to 32)

#### ➡️ Inputs 

None have been changed

#### ⬅️ Outputs 

None has been changed

##### Flu
- [TheiaCoV Illumina PE](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/f202eefb-1d81-4666-bd36-5bb324045464) 
- [TheiaCoV ONT](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/7743153e-fc54-4e5f-9525-72e8a19899a4)
- [TheiaCoV FASTA](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/0660fc70-e8da-41f9-9f59-19e43a0cb32b)

##### RSV
- [TheiaCoV Illumina PE](https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/ab584779-112f-4fb6-b81c-67cb69274ca2)

#### Suggested Scenarios for Reviewer to Test

- The container for VADR has been updated to the latest version so it's important to test all the organisms that are called to ensure that no major unintended changes occurred:
  - mpxv
  - sc2
  - wnv

## :microscope: Final Developer Checklist

- [ ] The workflow/task has been tested locally and results, including file contents, are as anticipated
- [x] The workflow/task has been tested on Terra and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (to be completed by Theiagen developer)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

## 🎯 Reviewer Checklist 

- [x] All impacted workflows/tasks have been tested on Terra with a different dataset than used for development
- [x] All reviewer-suggested scenarios have been tested and any additional
- [x] All changed results have been confirmed to be accurate
- [x] All workflows/tasks impacted by change/s have been tested using a standard validation dataset to ensure no unintended change of functionality
- [x] All code adheres to the style guide
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments

## 🗂️ Associated Documentation (to be completed by Theiagen developer)

- [ ] Relevant documentation on the Public Health Resources "PHB Main" has been updated
- [ ] Workflow diagrams have been updated to reflect changes

### CODE REVIEW

File: workflows/utilities/wf_organism_parameters.wdl [hunk 1]
```diff
@@ -76,8 +76,8 @@ workflow organism_parameters {
     Int flu_genome_len = 13000
 
     # vadr options are dummy options for flu right now
-    String flu_vadr_options = ""
-    Int flu_vadr_max_length = 0
+    String flu_vadr_options = "--atgonly --xnocomp --nomisc --alt_fail extrant5,extrant3 --mkey flu"
+    Int flu_vadr_max_length = 13600 # unsure about this value, by default they just use min length of 80
```
Comment (cimendes) [inline workflows/utilities/wf_organism_parameters.wdl:80]: change to 16000

Comment (cimendes) [inline workflows/utilities/wf_organism_parameters.wdl:80]: change to 13500

File: workflows/utilities/wf_organism_parameters.wdl [hunk 2]
```diff
@@ -76,8 +76,8 @@ workflow organism_parameters {
     Int flu_genome_len = 13000
```
Comment (cimendes) [inline workflows/utilities/wf_organism_parameters.wdl:76]: change to 13500

**Comment by kapsakcj** [comment]
Nice work on this! I need some more time looking at this, but so far I've looked at your VADR on Flu tests on TheiaCoV_Illumina_PE

Thoughts so far:
- VADR on Flu is running appropriately. I was cross referencing the VADR on Flu tutorial and everything looks as expected. 
  - Only 2 specific alerts: `extrant5,extrant3` will cause failure, but everything else is just a warning. So perhaps we communicate that to users that you should submit your sequences to NCBI as long as there are no outright failures (even if there are >0 warnings)
- code changes look great so far. TY for adding & exposing `vadr_memory`. CI is passing. So far I don't have any requested changes.

Curtis TODO: 

- [ ] review VADR on RSV outputs
- [ ] run SC2 samples through dev branch (ILMN PE wf) and ensure results are identical to using previous docker image with v1.4.2
- [ ] run Mpox samples through dev branch (FASTA wf)
- [ ] run WNV samples through dev branch (FASTA wf)

**Comment by kapsakcj** [comment]
Hmmmm I think we should enable the theiacov_Illumina_pe and se workflows in the CI for `miniwdl` and ensure they are updated. If we don't we're going to have CI failures after merging.

Similar to changes I've made in https://github.com/theiagen/public_health_bioinformatics/pull/375/files 

I'm happy to do this, but let me know if for some reason you think that is a bad idea.

**Comment by kapsakcj** [comment]
VADR on RSV appears to be running appropriately as well. There were a few samples that passed w/ 0 VADR alerts and some that failed with 2 VADR alerts (SRR19134713)

Here are some tests I ran:

- ✅ TheiaCov_FASTA on Flu and RSV (both A and B) samples: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/7ebe1167-a4a9-47ef-8f19-c0685f8f4b15
- ⚠️TheiaCoV_illumina_pe on RSV-A: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/68929b73-9e93-483f-b1b5-bc703b760d64
  - TODO - review outputs  
- ⚠️ TheiaCov_illumina_pe on sars-cov-2: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/fb847fc9-e6d8-4425-915e-fed2e60b1f91
  - TODO - review outputs
- ⚠️ TheiaCov_Illumina_se on sars-cov-2: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/e2e8f477-55d0-4f6b-b089-b5e5fb337b91
  - TODO - review outputs 
- ⚠️ TheiaCoV_FASTA on MPXV: https://app.terra.bio/#workspaces/theiagen-validations/curtis-sandbox-theiagen-validations/job_history/d2cd9cec-1b0e-48d6-928e-3a38063cb41f
  - TODO - review outputs
- ⚠️ TheiaCoV_illumina_pe on WNV: 
  - TODO - review outputs

**Review by kapsakcj — APPROVED** [review]
Once the tests succeed, I'll merge this PR. Nice work! You're a 🌟

**Pull request merged**